src/widget/wratedisplay.cpp

```diff
@@ -0,0 +1,42 @@
+#include "widget/wratedisplay.h"
+
+#include "control/controlobject.h"
+#include "control/controlproxy.h"
+#include "util/math.h"
+
+WRateDisplay::WRateDisplay(const QString& group, QWidget* parent)
+        : WNumber(parent) {
+    m_pRateRangeControl = new ControlProxy(group, "rateRange", this, ControlFlag::NoAssertIfMissing);
+    m_pRateRangeControl->connectValueChanged(this, &WRateDisplay::setValue);
+    m_pRateDirControl = new ControlProxy(group, "rate_dir", this, ControlFlag::NoAssertIfMissing);
+    m_pRateDirControl->connectValueChanged(this, &WRateDisplay::setValue);
+}
+
+void WRateDisplay::setup(const QDomNode& node, const SkinContext& context) {
+    WNumber::setup(node, context);
+
+    QDomElement rateDisplayPosition = context.selectElement(node, "Position");
+    m_nodePosition = rateDisplayPosition.text() == "Top" ? VerticalPosition::Top : VerticalPosition::Bottom;
+
```

Comment:
Re: split into prefix and range value
Here we could parse if we show prefix+value, prefix only or only the value.
Maybe with `<Display>[default|prefix|range]</Display>` > `m_displayFormat = [1|2|3]`? (if node is not present, empty or invalid show prefix+value)
Then use `m_displayFormat` in `setValue()` to decide what to display.

Reply:
Yes, this is a good idea, will implement the <Display> attribute, will also create enum for the display format